src/pages/Share.js

```diff
@@ -0,0 +1,140 @@
+  };
+
+  const handlePathSelectionClick = async () => {
+    exportPath(selectedPathName).then((serializedPath) => {
+      setSelectedPath(serializedPath);
+    });
+  };
+
+  const handleQRScan = async (result) => {
+    console.log(result);
+    if (result.startsWith(QRCODE_PREFIX)) {
+      result.substring();
+      const id = result.slice(QRCODE_PREFIX.length);
+      setTargetPeerID(id);
+      await connectToPeerAndReceive(peer, targetPeerID, importPath);
+    } else {
+      console.warn('Unknown QR code');
+    }
+  };
+
+  const handleQRScanError = (error) => {
+    console.error(error);
+  };
+
+  useEffect(() => {
+    // Check for camera availability
+    navigator.mediaDevices
+      .getUserMedia({ video: true })
+      .then(() => {
+        setIsCameraAvailable(true);
+      })
+      .catch((err) => {
+        console.error('Camera not available:', err);
+      });
+  }, []);
+
+  useEffect(() => {
+    // Initialize WebRTC
+    const initPeer = async () => {
+      const { id: newId, peer: newPeer } = await initializePeer();
+      setPeerId(newId);
+      setPeer(newPeer);
+    };
+    if (!peer) {
+      initPeer();
+    }
+
+    return () => {
+      if (peer) {
+        peer.destroy();
+      }
+    };
+  }, [peer]);
+
+  useEffect(() => {
+    const handleNewConnection = async () => {
+      console.log('Setting up onConnection with', peer, selectedPath);
+      sendDataOnConnection(peer, selectedPath)
+        .then(() => console.log('Successfully sent path', selectedPath))
+        .catch((e) => {
+          console.log('Error while connecting and sending path', e);
+        });
+    };
+
+    if (!(peer && selectedPath)) {
+      return;
+    }
+
+    handleNewConnection();
+  }, [peer, selectedPath]);
+
+  return (
+    <div>
+      <div className="sender-container">
+        <div className="path-selection">
+          <label>
+            Jaettavan polun nimi
+            <input
+              type="text"
+              value={selectedPathName}
+              onChange={(e) => setSelectedPathName(e.target.value)}
+            />
+          </label>
+          <button onClick={handlePathSelectionClick}>Valitse polku</button>
+        </div>
+        {peerId ? (
+          <div>
+            <label>Jakajan ID:</label>
+            <span style={{ marginLeft: '10px' }}>{peerId}</span>
+            <QRCode value={QRCODE_PREFIX + peerId} />
+          </div>
+        ) : (
+          <label>Yhdistetään...</label>
+        )}
+      </div>
+
+      <div style={{ marginTop: '100px' }} className="receiver-container">
+        <div>
+          <label>Syötä jakajan ID</label>
+          <input
+            style={{ marginLeft: '10px' }}
+            type="text"
+            value={targetPeerID}
+            onChange={(e) => setTargetPeerID(e.target.value)}
+          />
+          {isCameraAvailable ? (
+            <Scanner onScan={handleQRScan} onError={handleQRScanError} />
```

Comment:
Läppärillä kun annan luvan kameralle niin näkymä on tämä. Taitaa puhelimella toimia paremmin, mutta vähän vaikea kai testata

![image](https://github.com/user-attachments/assets/517405c6-f93d-4ae7-87c3-6ec99e527e95)

Reply:
Joo tuo sama näkymä tulee jos renderöi komponentin ja ei ole kameraa. En edes testannut missään ympäristössä jossa olisi kamera, ajatuksena oli toivoa parasta että tuo komponentti on plug and play ja toimii :D
Tutkin ja testailen itsekin vielä läppärillä jos saisin tuon jotenkin toimimaan järkevästi. Ehkä myös tuota layout voisin koittaa saada paremmaksi, tuo "Hae polku" nappi katoaa jonnekin kun tuo kamera näkymä aukeaa.

Ajattelin että testailee lähinnä mobiilinäkymässä QR koodeilla aluksi, sellaista manuaalista testailua pitkälti. Jotain yksikkötestejä tuolle kirjoittelen sitten kun on tiedossa että testataan edes oikeita asioita.

Reply:
Nyt on hyvä! ✨